paper.md

```diff
@@ -0,0 +1,95 @@
+This is particularly useful for foundational libraries on which the scientific software ecosystem is built,
+but any researcher who tests their software in Python can benefit from these facilities. 
+
+For example, it has found bugs in astropy ([@astropy:2018]) and numpy ([@DBLP:journals/cse/WaltCV11]),
+both of which are foundational libraries on which a significant amount of research is based.
+In particular it has proven very effective at finding numerical stability issues in such software.
+
+Additionally, Hypothesis is easily extensible, and has a number of user-provided exceptions for specific research applications.
+For example, hypothesis-networkx^[https://pypi.org/project/hypothesis-networkx/] generates graph data structures,
+and hypothesis-bio^[https://pypi.org/project/hypothesis-bio/] generates formats suitable for bioinformatics.
+As it is used by more researchers, the number of research applications will only increase.
+
+By lowering the barrier to effective testing, Hypothesis makes testing of research software written in Python much more compelling,
+and has the potential to significantly improve the quality of the associated scientific research as a result.
+
+# Hypothesis for Software Testing Research
+
+This is true both because of the much wider array of software that can now easily be tested with property-based testing,
+and because it has a novel implementation that lifts a major difficulty that prior research on software testing faces.
+
+Much of software testing research boils down to variants on the following problem:
+Given some interestingness test (e.g. that it triggers a bug in some software),
+how do we generate a "good" test case that passes that interestingness test?
+
+Particular subproblems of this are:
+
+1. How do we generate test cases that satisfy difficult interestingness tests?
+2. How do we ensure we generate only valid test cases? (the *test-case validity problem* - see @DBLP:conf/pldi/RegehrCCEEY12)
+3. How do we generate human readable test cases?
+ 
+Traditionally property-based testing has adopted random test-case generation to find interesting test cases,
+followed by test-case reduction (see [@DBLP:conf/pldi/RegehrCCEEY12, DBLP:journals/tse/ZellerH02]) to turn them into human readable ones,
+requiring the users to manually specify a *validity oracle* (a predicate that identifies if an arbitrary test case is valid) to avoid invalid test cases.
+
+The chief limitations of this from a user's point of view are that writing correct validity oracles is difficult and annoying,
+and that random generation, while often much better than hand-written examples, is not especially good at satisfying difficult properties.
+Fortunately it is very easy to define correct random generators,
+and the users can then defer the rest of the work until their tests start finding bugs, when they are more likely to be willing to do the work.
```

Comment:
I'd split this into two paragraphs, and extend the first part:  Because of the difficulty of specifying a validity oracle in addition to a test-case generator, traditional approaches have seen very little adoption in research domains beyond software development of e.g. compilers.

Reply:
I don't think that's accurate. Traditional approaches have seen *loads* of uptake, that's what property-based testing is! Way more people are using PBT than are testing compilers.